prime-router/src/main/kotlin/history/azure/DatabaseSubmissionsAccess.kt

```diff
@@ -38,6 +42,77 @@ class DatabaseSubmissionsAccess(
         return senderFilter
     }
 
+    /**
+     * Fetch a single (usually detailed) action of a specific type.
+     *
+     * @param actionId the action id attached to this submission.
+     * @param klass the class that the found data will be converted to.
+     * @return the submission matching the given query parameters, or null.
+     */
+    override fun <T> fetchAction(
+        actionId: Long,
+        klass: Class<T>
+    ): T? {
+        return db.transactReturning { txn ->
+            DSL.using(txn)
+                .select(detailedSelect())
+                .from(ACTION)
+                .where(
+                    ACTION.ACTION_ID.eq(actionId)
+                )
+                .fetchOne()?.into(klass)
+        }
+    }
+
+    /**
+     * Add logs and reports related to the history being fetched.
+     *
+     * @return a jooq select statement adding additional DB columns.
+     */
+    fun detailedSelect(): List<SelectFieldOrAsterisk> {
```

Comment:
Not really liking how this is used - it may need to be refactored. It is only called from one place, but will generate an inefficient subquery when run. Lets pair on this.